core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -127,6 +128,16 @@
     public static final Setting<Integer> MAX_SLICES_PER_SCROLL = Setting.intSetting("index.max_slices_per_scroll",
         1024, 1, Property.Dynamic, Property.IndexScope);
 
+    /**
+     * The number of shard copies required to be active before returning from an index creation operation.
+     */
+    public static final Setting<ActiveShardCount> WAIT_FOR_ACTIVE_SHARDS_SETTING =
+        new Setting<>("index.wait_for_active_shards",
+                         (String) null,
```

Comment:
we could also add the default here: "1". This would simplify the "resolve" code a bit.